src/test/java/com/google/vinet/servlets/RequestServletTest.java

```diff
@@ -0,0 +1,281 @@
+package com.google.vinet.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.datastore.DatastoreFailureException;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import javax.servlet.Registration;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+public class RequestServletTest {
```

Comment:
Can you do a test to ensure subjects and details which are input are also the same as subjects and details which are stored in datastore, to ensure the ordering isn't accidentally changes since the index is used as the identifier for which subject matches which details?

Reply:
I've fixed this now https://github.com/googleinterns/step226-2020/pull/8/commits/f7b9f4606da3c2315c014d264f2ffb44cfc4e8ce